src/test/java/bio/overture/ego/service/RefreshContextServiceTest.java

```diff
@@ -36,11 +37,34 @@
   @Rule public ExpectedException exceptionRule = ExpectedException.none();
 
   @Test
-  public void refresh_usedTokenIsDeleted() {
+  public void refresh_invalidContext_usedTokenIsDeleted() {
+    val user1 = entityGenerator.setupUserWithRefreshToken("User One");
+    // force invalid refreshContext with invalid user status
+    user1.setStatus(StatusType.PENDING);
+    val user1Token = tokenService.generateUserToken(user1);
+    val refreshToken1 = user1.getRefreshToken();
+
+    Assert.assertTrue(refreshContextService.getById(refreshToken1.getId()) != null);
```

Comment:
this assertion is not enough. A better one is:
```java
assertTrue(refreshContextService.getById(refreshToken1.getId) != null);
assertTrue(refreshContextService.getById(refreshToken1.getId()).isPresent());
```
Even just doing the isPresent check is enough for a test. In general, if an `Optional<>` object is returned then you can assume the developer intends it to always be not null. you can have empty optional objects, and non-empty optional objects. Returning null on top of that defeats the purpose of Optional :)